src/main/java/io/jenkins/plugins/pipeline_elasticsearch_logs/write/fluentd/FluentdWriter.java

```diff
@@ -233,7 +235,11 @@ private void emitData(String tag, Map<String, Object> data) throws IOException {
         while (true) {
             LOGGER.log(Level.FINEST, "Emitting data: Try {0} Data: {1}", new Object[] {count, data });
             try {
-                fluentd.emit(tag, data);
+                Instant instant = Instant.parse((String) data.get("timestamp"));
+                long epochSeconds = instant.getEpochSecond();
+                long nanoSeconds = instant.getNano();
+                EventTime eventTime = EventTime.fromEpoch(epochSeconds, nanoSeconds);
```

Comment:
Move this lower-level code into a separate method, taking `data` as input and returning `EventTime`.

Reply:
Done in [c181d8a](https://github.com/SAP/elasticsearch-logs-plugin/pull/96/commits/c181d8a6ace2d0f9e1b3662c715721afe9158dd3).